cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -127,20 +133,21 @@ impl Cmd {
     }
 }
 
+#[derive(RustEmbed)]
+#[folder = "src/utils/contract-init-template"]
+struct TemplateFiles;
+
 fn init(
     project_path: &Path,
     frontend_template: &String,
     with_examples: &[String],
 ) -> Result<(), Error> {
-    let cli_cmd_root = env!("CARGO_MANIFEST_DIR");
-    let template_dir_path = Path::new(cli_cmd_root)
-        .join("src")
-        .join("utils")
-        .join("contract-init-template");
-
     // create a project dir, and copy the contents of the base template (contract-init-template) into it
-    std::fs::create_dir_all(project_path)?;
-    copy_contents(template_dir_path.as_path(), project_path)?;
+    std::fs::create_dir_all(project_path).map_err(|e| {
+        eprintln!("Error creating new project directory: {project_path:?}");
```

Comment:
Not sure if this is overkill, but I ended up printing out a message in addition to the error message for a bunch of potential errors, since a lot of them end up being pretty generic `IO` errors, and it may be difficult to track down the origin of the error. 

Happy to rethink this if folks have some other ideas!

Reply:
I think this is fine for now.  It would be nice if you were using init as a library to have the more specific error, but most CLI's use anyhow, which acts similar to this.

Reply:
Interesting! Are we using any other commands as libraries at the moment?

Reply:
It's used  by the soroban-test crate. But I've used it a lot with other CLI projects. Makes it easy to plug an existing CLI into a subcommand and link them all statically.

Reply:
Cool, i'll take a look!